scripts/release/asset_publish.py

```diff
@@ -304,6 +374,30 @@ def get_asset_details(
     return json.loads(result.stdout)
 
 
+def get_parsed_details_from_asset_uri(asset_type: str, asset_uri: str) -> Tuple[str, str, str, str]:
+    """Validate asset URI and return parsed details. Exception is raised for an invalid URI.
+
+    :param asset_type: Valid values are component, environment and model
+    :type asset_type: str
+    :param asset_uri: A workspace or registry asset URI to parse
+    :type asset_uri: str
+    :return:
+        A tuple with asset `name`, `version`, `label`, and `registry_name` in order.
+        `label` and `registry_name` will be None for workspace URI.
+    :rtype: Tuple
+    """
+    REGISTRY_ASSET_PATTERN = re.compile(rf"{REGISTRY_ASSET_STR.format(asset_type)}")
+    asset_registry_name = None
+    if (match := REGISTRY_ASSET_PATTERN.match(asset_uri)) is not None:
+        asset_registry_name, asset_name, asset_version, asset_label = (
+            match.group(1), match.group(2), match.group(3), match.group(4))
+    elif (match := WORKSPACE_ASSET_PATTERN.match(asset_uri)) is not None:
+        asset_name, asset_version, asset_label = match.group(1), match.group(2), match.group(3)
```

Comment:
See https://docs.python.org/3/library/re.html#re.Match.groups
```suggestion
        asset_name, asset_version, asset_label = match.groups()
```